core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -1135,4 +1147,69 @@ public synchronized void updateGroup() {
       mRaftGroup = newGroup;
     }
   }
+
+  private RaftProtos.RoleInfoProto getRaftRoleInfo() {
+    GroupInfoReply groupInfo = null;
+    try {
+      groupInfo = getGroupInfo();
+    } catch (IOException e) {
+      LOG.error("Error while get group info", e);
+    }
+    if (groupInfo == null || groupInfo.getException() != null) {
+      return null;
+    }
+    return groupInfo.getRoleInfoProto();
+  }
+
+  /**
+   * Get the role name.
+   *
+   * @return the role name
+   */
+  public int getRoleId() {
+    RaftProtos.RoleInfoProto roleInfo = getRaftRoleInfo();
+    if (roleInfo != null) {
+      return roleInfo.getRoleValue();
+    } else {
+      return -1;
+    }
+  }
+
+  /**
+   * Get the leader id.
+   *
+   * @return the leader id
+   */
+  public String getLeaderId() {
+    RaftProtos.RoleInfoProto roleInfo = getRaftRoleInfo();
+    if (roleInfo == null) {
+      return null;
```

Comment:
`getLeaderIndex()` will NPE if you return null

Reply:
Thanks, return UNDEFINED_LEADER_ID